truffleruby/src/main/ruby/core/truffle/cext.rb

```diff
@@ -1640,6 +1614,89 @@ def rb_tr_log_warning(message)
     Truffle::Debug.log_warning message
   end
 
+  class RData
+
+    DATA_FIELD_INDEX = 2
+
+    def initialize(object)
+      @object = object
+    end
+
+    def [](index)
+      raise unless index == DATA_FIELD_INDEX
+      @object.instance_variable_get(:@data)
+    end
+
+    def []=(index, value)
+      raise unless index == DATA_FIELD_INDEX
+      @object.instance_variable_set :@data, value
+    end
+
+  end
+
+  def RDATA(object)
+    RData.new(object)
+  end
+
+  class RbEncoding
+
+    NAME_FIELD_INDEX = 0
+
+    attr_reader :encoding
+
+    def initialize(encoding)
+      @encoding = encoding
+    end
+
+    def [](index)
+      raise unless index == NAME_FIELD_INDEX
+      @encoding.name
+    end
+
+  end
+
+  def rb_to_encoding(encoding)
+    encoding = Encoding.find(encoding.to_str) unless encoding.is_a?(Encoding)
+    RbEncoding.new(encoding)
+  end
+
+  def rb_enc_from_encoding(rb_encoding)
+    rb_encoding.encoding
+  end
+
+  class RStringPtr
+
+    attr_reader :string
+
+    def initialize(string)
+      @string = string
+    end
+
+    def size
+      Truffle::CExt::string_pointer_size(@string)
```

Comment:
These should be `.` method calls.
Even though it's also valid Ruby syntax it's deprecated (like Math::cos vs Math.cos).